### PR TITLE
Support sys.dm_os_sys_info view

### PR DESCRIPTION
**Pull request by sharathbp**
### Description

1. With this change we support sys.dm_os_sys_info view which provides the following information
- ms_ticks - Specifies the number of milliseconds since the computer started. Not nullable.
- sqlserver_start_time - Specifies the local system date and time Engine last started. Not nullable.
- sqlserver_start_time_ms_ticks - Represents the ms_tick number when Engine last started. Compare to the current ms_ticks column. Not nullable.

2. sys.dm_os_sys_info view is used by RDSProxy for fetching DB Uptime. UPTIME is required to detect DB restart by comparing with previous UPTIME value which is stored by RDSProxy. If the current UPTIME < prev UPTIME means DB would have restarted. So if the DB is restarted RDSProxy have to re-establish the connection as the old connection is lost. 

### Issues Resolved
JIRA: BABEL-4824

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** Yes

* **Arbitrary inputs -** Yes

* **Negative test cases -** Yes

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 1]
```diff
@@ -3601,3 +3601,10 @@ AS SELECT
 FROM sys.objects so
 WHERE FALSE;
 GRANT SELECT ON sys.sequences TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.dm_os_sys_info 
+AS SELECT 
+  CAST(ROUND(EXTRACT(EPOCH FROM NOW()) * 1000.0, 0) AS BIGINT) AS ms_ticks,
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:3608]: does this server our purpose? how do we track server start time?

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 2]
```diff
@@ -3601,3 +3601,10 @@ AS SELECT
 FROM sys.objects so
 WHERE FALSE;
 GRANT SELECT ON sys.sequences TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.dm_os_sys_info
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:3605]: according to TSQL doc, this view is supposed to contain many more columns

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 1 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3608]: Yes, we get server start time from pg_postmaster_start_time() below. To get the uptime Proxy will do (ms_ticks -sqlserver_start_time_ms_ticks).

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 2 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3605]: Yes other are related to hardware and os informations. For supporting Proxy this below added columns are sufficient. I was thinking of add these fields now and add the other fields as and when required.

Comment (thephantomthief) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3605]: I think we should add all columns right now. We can put dummy values for the columns that do not apply to us and we can update those later if we ever find suitable values for them. Changing values in upgrade scripts is very straightforward but adding columns, not so much.

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 3]
```diff
@@ -3601,3 +3601,44 @@ AS SELECT
...
+  CAST(ROUND(EXTRACT(EPOCH FROM pg_postmaster_start_time()) * 1000.0, 0) AS BIGINT) AS sqlserver_start_time_ms_ticks, 
+  CAST(pg_postmaster_start_time() AS sys.DATETIME) AS sqlserver_start_time,
+  CAST(0 AS INT) AS affinity_type,
+  CAST(NULL AS sys.NVARCHAR(60)) AS affinity_type_desc,
+  CAST(0 AS BIGINT) AS process_kernel_time_ms,
+  CAST(0 AS BIGINT) AS process_user_time_ms,
+  CAST(0 AS INT) AS time_source,
+  CAST(NULL AS sys.NVARCHAR(60)) AS time_source_desc,
+  CAST(0 AS INT) AS virtual_machine_type,
+  CAST(NULL AS sys.NVARCHAR(60)) AS virtual_machine_type_desc,
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/sys_views.sql]: This can be `'NONE'` instead of NULL

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 4]
```diff
@@ -3601,3 +3601,44 @@ AS SELECT
...
+  CAST(0 AS INT) AS softnuma_configuration,
+  CAST(NULL AS sys.NVARCHAR(60)) AS softnuma_configuration_desc,
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/sys_views.sql]: This can be `'OFF'` instead of NULL

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 2 above)
Comment (Deepesh125) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3605]: +1

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 5]
```diff
@@ -3601,3 +3601,44 @@ AS SELECT
...
 WHERE FALSE;
 GRANT SELECT ON sys.sequences TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.dm_os_sys_info 
+AS SELECT 
+  CAST(0 AS BIGINT) AS cpu_ticks,
+  CAST(ROUND(EXTRACT(EPOCH FROM NOW()) * 1000.0, 0) AS BIGINT) AS ms_ticks, 
+  CAST(0 AS INT) AS cpu_count,
+  CAST(0 AS INT) AS hyperthread_ratio,
+  CAST(0 AS BIGINT) AS physical_memory_kb,
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:3611]: physical_memory_in_bytes is still missing

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 6]
```diff
@@ -3601,3 +3601,44 @@ AS SELECT
...
+  CAST(0 AS BIGINT) AS virtual_memory_kb,
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:3612]: virtual_memory_in_bytes is missing

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 7]
```diff
@@ -3601,3 +3601,44 @@ AS SELECT
...
+  CAST(0 AS BIGINT) AS committed_kb,
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:3613]: bpool_committed is missing

Comment (Deepesh125) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3613]: and there are still few more columns missing, Please double check

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 5 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3611]: the doc says its for **Applies to: SQL Server 2008 (10.0.x) and SQL Server 2008 R2 (10.50.x).** so not added it. I also verified with latest sqlserver 16 and it not present. I think its not required.

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 7 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3613]: Yeah I have double checked. The other fields are for older versions and not supported in sqlserver 16 so not added them.

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 6 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3612]: Not required in sqlserver 16

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 3 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:3633]: I checked the RDS sqlserver and it specifies **virtual_machine_type_desc** as HYPERVISOR. Since APG also runs in AL2 virtual machine I think it should be HYPERVISOR to be precise.

**Review by Deepesh125 — APPROVED** [review]

**Review by thephantomthief — APPROVED** [review]

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 4 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql]: fixed

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 3 above)
Comment (sharathbp) [reply contrib/babelfishpg_tsql/sql/sys_views.sql]: fixed
